File: app/src/main/java/com/example/wpossbank/TransfersActivity.java
package com.example.wpossbank;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.example.wpossbank.fragments.Dialogs;
import com.example.wpossbank.modelos.Admin;
import com.example.wpossbank.modelos.LogEntry;
import com.example.wpossbank.modelos.MakeMessages;
import com.example.wpossbank.modelos.Validate;

import eightbitlab.com.blurview.BlurView;
import eightbitlab.com.blurview.RenderScriptBlur;

public class TransfersActivity extends AppCompatActivity {
    Context context;
    Validate validate;

    Admin admin;
    MakeMessages messages;

    BlurView blurView;

    EditText ccInput, pinInput, pinConfirmInput, ccTransferInput, transferInput;
    Button goBackButton, confirmButton;
    ImageView backArrow;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_transfers);
        context = this;
        validate = new Validate(context);

        admin = new Admin();
        messages = new MakeMessages();

        blurView = findViewById(R.id.blurView);
        blurBackground();

        ccInput = findViewById(R.id.ccInput);
        pinInput = findViewById(R.id.pinInput);
        pinConfirmInput = findViewById(R.id.pinConfirmInput);
        ccTransferInput = findViewById(R.id.ccTransferInput);
        transferInput = findViewById(R.id.transferInput);

        confirmButton = findViewById(R.id.confirmButton);
        backArrow = findViewById(R.id.backArrow);

        confirmButton.setOnClickListener(confirmDeposit -> {
            //Deckara y verifica si los campos tienen la información correcta
            boolean ccValidate = validate.isNumber(ccInput) && validate.matchUserData(ccInput, "cc"),
                    pinValidate = validate.pin(pinInput) && validate.matchUserData(pinInput, "pin"),
                    ccTransferValidate = validate.isInRange(ccTransferInput,10,13) &&
                            validate.isInDatabase(ccTransferInput, "user", "cc") &&
                            validate.notMatchUserData(ccTransferInput, "cc"),
                    transferValidate = validate.useBalance(transferInput);

            if (!validate.isEmpty(pinConfirmInput) &&
                    !pinInput.getText().toString().equals(pinConfirmInput.getText().toString())) {
                Log.d("TRANSFER", "The pins do not match, " +
                        "pin="+pinInput.getText().toString() +" confirm="+pinConfirmInput.getText().toString());

                pinConfirmInput.setError(getResources().getString(R.string.error_wrong));
            }else if (ccValidate && pinValidate && ccTransferValidate && transferValidate){
                admin.setBalance(admin.getCost()/2);

                LogEntry logEntry = new LogEntry();
                logEntry.setType("transfer");
                logEntry.setAmount(Integer.parseInt(transferInput.getText().toString()));
                logEntry.setSource(ccInput.getText().toString());

                new Dialogs.ConfirmUserTransferBalance(context , logEntry, admin, ccTransferInput.getText().toString(),
                        messages.transfer(context, transferInput, ccTransferInput))
                        .show(getSupportFragmentManager(),"CONFIRM");
            }else {
                // Avisa al usuario si hay un campo con valores invalidos
                Toast.makeText(context, R.string.error_invalid_input, Toast.LENGTH_LONG).show();
            }
        });

        backArrow.setOnClickListener( goBack -> finish());
    }

    private void blurBackground(){
        float radius = 20f;

        View decorView = getWindow().getDecorView();
        ViewGroup rootView = decorView.findViewById(android.R.id.content);
        Drawable windowBackground = decorView.getBackground();

        blurView.setupWith(rootView)
                .setFrameClearDrawable(windowBackground)
                .setBlurAlgorithm(new RenderScriptBlur(this))
                .setBlurRadius(radius)
                .setBlurAutoUpdate(true)
                .setHasFixedTransformationMatrix(true);
    }
}

File: app/src/main/java/com/example/wpossbank/activities/TransfersActivity.kt
package com.example.wpossbank.activities

import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.wpossbank.R
import com.example.wpossbank.fragments.Dialogs.ConfirmUserTransferBalance
import com.example.wpossbank.models.Admin
import com.example.wpossbank.models.LogEntry
import com.example.wpossbank.models.MakeMessages
import com.example.wpossbank.models.Validate
import eightbitlab.com.blurview.BlurView
import eightbitlab.com.blurview.RenderScriptBlur

class TransfersActivity : AppCompatActivity() {
    lateinit var context: Context
    lateinit var validate: Validate
    lateinit var admin: Admin
    lateinit var messages: MakeMessages
    lateinit var blurView: BlurView
    lateinit var ccInput: EditText
    lateinit var pinInput: EditText
    private lateinit var pinConfirmInput: EditText
    private lateinit var ccTransferInput: EditText
    private lateinit var transferInput: EditText
    lateinit var confirmButton: Button
    lateinit var backArrow: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_transfers)
        context = this

        validate = Validate(context)
        admin = Admin()
        messages = MakeMessages()

        blurView = findViewById(R.id.blurView)
        blurBackground()

        ccInput = findViewById(R.id.ccInput)
        pinInput = findViewById(R.id.pinInput)
        pinConfirmInput = findViewById(R.id.pinConfirmInput)
        ccTransferInput = findViewById(R.id.ccTransferInput)
        transferInput = findViewById(R.id.transferInput)
        confirmButton = findViewById(R.id.confirmButton)
        backArrow = findViewById(R.id.backArrow)

        confirmButton.setOnClickListener {
            //Deckara y verifica si los campos tienen la información correcta
            val ccValidate = validate.isNumber(ccInput) && validate.matchUserData(ccInput, "cc")
            val pinValidate = validate.pin(pinInput) && validate.matchUserData(pinInput, "pin")
            val ccTransferValidate = validate.isInRange(ccTransferInput, 10, 13) &&
                    validate.isInDatabase(ccTransferInput, "user", "cc") &&
                    validate.notMatchUserData(ccTransferInput, "cc")
            val transferValidate = validate.useBalance(transferInput)
            if (!validate.isEmpty(pinConfirmInput) &&
                pinInput.text.toString() != pinConfirmInput.text.toString()
            ) {
                Log.d(
                    "TRANSFER", "The pins do not match, " +
                            "pin=" + pinInput.text
                        .toString() + " confirm=" + pinConfirmInput.text.toString()
                )
                pinConfirmInput.error = resources.getString(R.string.error_wrong)
            } else if (ccValidate && pinValidate && ccTransferValidate && transferValidate) {
                admin.balance = admin.cost / 2
                val logEntry = LogEntry()
                logEntry.type = "transfer"
                logEntry.amount = transferInput.text.toString().toInt()
                logEntry.source = ccInput.text.toString()
                ConfirmUserTransferBalance(logEntry, admin, ccTransferInput.text.toString(),
                    messages.transfer(context, transferInput, ccTransferInput))
                    .show(supportFragmentManager, "CONFIRM")
            } else {
                // Avisa al usuario si hay un campo con valores invalidos
                Toast.makeText(context, R.string.error_invalid_input, Toast.LENGTH_LONG).show()
            }
        }
        backArrow.setOnClickListener { finish() }
    }

    private fun blurBackground() {
        val radius = 20f
        val decorView = window.decorView
        val rootView = decorView.findViewById<ViewGroup>(android.R.id.content)
        val windowBackground = decorView.background
        blurView.setupWith(rootView)
            .setFrameClearDrawable(windowBackground)
            .setBlurAlgorithm(RenderScriptBlur(this))
            .setBlurRadius(radius)
            .setBlurAutoUpdate(true)
            .setHasFixedTransformationMatrix(true)
    }
}